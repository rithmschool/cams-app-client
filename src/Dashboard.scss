@import "variables";

.email-sent,
.loading {
  margin-bottom: 30px;

  div {
    display: inline-block;
  }

  i {
    margin-left: 10px;
  }
}

/*PLAYLIST
*********/

.playlist-container {
  border: 3px solid #009eb8;
  border-radius: 10px;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  flex-wrap: wrap;
  background: $light-grey;
  padding: 1rem;
}
#playlist-name {
  width: 75%;
}
input[type="checkbox"] + .toggle {
  display: none;
}
input[type="checkbox"]:checked + .toggle {
  display: block;
  text-align: left;
}
/*.thead {
  display: flex;
  justify-content: row;
}
.thead span{
  display: flex;
  flex-grow: 1;
  justify-content: center;
  border: 3px solid #00aea7;
  color: #00c5cc;
}
.thead span::first-child{
  border-top-right-radius: 7px;
}
.thead span::first-child{
  border-top-right-radius: 7px;
}*/

/*.playlist-card-contents {
  background: white;
  border-radius: 0.2rem;
  margin: 0.4rem;
  padding: 0.5rem;
  box-shadow: $nav-shadow;
}

.playlist-card-contents:focus {
  outline: 0px;
}
*/
.email-form {
  align-self: center;
}

.selected {
  border: 0.2px solid $success;
}

.hidden {
  display: none;
}

.video-title {
  padding: 0.5rem;
  color: $bold-grey;
  text-align: left;
  max-width: 15rem;
  font-size: 12px;
}

.delete {
  color: rgb(208, 208, 208);
}

.delete:hover {
  color: $danger;
}
