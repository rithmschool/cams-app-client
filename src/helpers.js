export const BASE_URL = process.env.REACT_APP_SERVER_URL || 'http://localhost:5000'

export const ensureCorrectUser = (pathId) => {
  let token = localStorage.getItem('token');
  if(!token){return false;}
  let tokenId = JSON.parse(atob(localStorage.getItem('token').split('.')[1])).id
  return +tokenId === +pathId;
}

export const userID = () => JSON.parse(atob(localStorage.getItem('token').split('.')[1])).id

export const config = {
  headers: {
    'Accept':'application/json',
    'ContentType':'application/json',
    'Authorization':'bearer ' + localStorage.getItem('token')
  }
}
