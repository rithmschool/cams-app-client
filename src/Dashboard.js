import React, {Component} from 'react';
import {BASE_URL} from './helpers.js';
import axios from 'axios';
import {Link} from 'react-router-dom'

class Dashboard extends Component{

	constructor(props) {
		super(props)
		this.state = {
			email: "",
			playlistID: null,
			playlistName: null,
      userPlaylists: [],
		}
		this.closeSelection = this.closeSelection.bind(this)
	}

	sendMail(config, thisArg) {
		axios.post(`${BASE_URL}/api/users`, {
			email: thisArg.state.email
		}, config)
		.then(response => axios.post(`${BASE_URL}/api/assessments`, {
			patient_id: response.data.id,
			playlist_id: thisArg.state.playlistID
			}, config))
		.then(response => axios.post(`${BASE_URL}/api/users/mail`, {
			assessment_id: response.data.id,
			patient_id: response.data.patient_id
			}, config))
	}

	handleChange(e) {
		this.setState({
			[e.target.name]: e.target.value
		});
	}

	handleSubmit(e) {
		let config = {
			headers: {
				'Accept': 'application/json',
				'Content-Type': 'application/json',
				'Authorization': 'bearer ' + localStorage.getItem('token')
			}
		}
		e.preventDefault()
		this.sendMail(config, this)
	}

	closeSelection() {
		this.setState({playlistID: null, playlistName: null})
	}

	choosePlaylist(playlist_id, playlistName){
		if (this.state.playlistID === null || playlist_id !== this.state.playlistID){
			this.setState({playlistID: playlist_id, playlistName: playlistName})
		}
	}

  componentWillMount(){
    let userId = JSON.parse(atob(localStorage.getItem('token').split('.')[1])).id
    axios.get(
      `${BASE_URL}/api/users/${userId}/playlists`,
      {
        headers: {
        	'Accept':'application/json',
          'ContentType':'application/json',
          'Authorization':'bearer ' + localStorage.getItem('token')
        }
      }
    ).then(response => {
      this.setState({userPlaylists: response.data})
    })
  }

  render() {
    let playlists = this.state.userPlaylists.map((playlist, i) => {
			let showForm = this.state.playlistName === playlist.name ?
				<div>
					<form onSubmit={this.handleSubmit.bind(this)}>
						<h5>Send to:</h5>
						<input
							type="email"
							name="email"
							placeholder="email"
							value={this.state.email}
							required onChange={this.handleChange.bind(this)}
						/>
						<button
							className="button button-hover"
							type="submit"
							value="Submit">
							Submit
						</button>
					</form>
					<div onClick={this.closeSelection}>
						<i className="fa delete fa-times-circle button-hover" aria-hidden="true"></i>
					</div>
				</div> :
				null
			let className =
				this.state.playlistID === playlist.id ?
					'selected' :
					'playlist-card'
			return(
				<div
					key={i}
					tabIndex="0"
					className={`${className} button-hover`}
					onClick={this.choosePlaylist.bind(this, playlist.id, playlist.name)}
				>
					<h5 className="playlist-name-title">{playlist.name}</h5>
					{playlist.videos.map((video, idx) => {
						return(
							<p className="song-title" key={idx} >
								{video.title}
							</p>
						)
					})
				}
				{showForm}
			</div>
			)
		})
		return (
			<div>
        <div className="banner-text">
          <h1 className="banner-bold">Dashboard</h1>
        </div>
        <div className="content">
          <div className="dash-nav">
            <Link to="/playlists/new">
              <button
                className="button button-hover big-nav"
                type="submit"
                value="Submit">
                  New Playlist
              </button>
            </Link>
						<Link to="/assessments">
							<button
								className="button button-hover big-nav"
								type="submit"
								value="Submit">
									Assessments
							</button>
						</Link>
          </div>
					<div className="playlist-card">
						{playlists}
					</div>
        </div>
      </div>
		)
	}
}

export default Dashboard;
