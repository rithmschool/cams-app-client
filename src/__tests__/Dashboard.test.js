import React from "react";
import ReactDOM from "react-dom";
import { shallow, mount } from "enzyme";
import renderer from "react-test-renderer";
import Dashboard from "../Dashboard";
import toJson from "enzyme-to-json";
import { Switch, MemoryRouter, Link } from "react-router-dom";
import MockAdapter from "axios-mock-adapter";
import { BASE_URL, userID, config } from "../helpers.js";
import LocalStorageMock from "../setupTests/LocalStorageMock";
import axios from "axios";

var mock = new MockAdapter(axios);

beforeAll(() => {
  localStorage.setItem("token", ".eyJpZCI6NH0=.");
  mock.onGet().reply(200, []);
});

test("dashboard matches snapshot ", () => {
  var dashboard = shallow(<Dashboard />);
  expect(toJson(dashboard)).toMatchSnapshot();
});

test("renders without crashing", () => {
  shallow(<Dashboard />);
});

test("mockAdapter works", () => {
  const dashboard = shallow(<Dashboard />);
  dashboard.setState({
    userPlaylists: [
      {
        id: 1,
        name: "playlist 1",
        videos: []
      },
      {
        id: 2,
        name: "playlist 2",
        videos: []
      }
    ]
  });
  expect(toJson(dashboard)).toMatchSnapshot();
});

afterAll(() => {
  mock.restore();
});
