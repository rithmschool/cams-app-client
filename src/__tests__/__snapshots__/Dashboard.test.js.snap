// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dashboard matches snapshot  1`] = `
<div>
  <div
    className="banner-text"
  >
    <h1
      className="banner-bold"
    >
      Dashboard
    </h1>
  </div>
  <div
    className="content"
  >
    <div
      className="playlist-container"
    />
  </div>
</div>
`;

exports[`mockAdapter works 1`] = `
<div>
  <div
    className="banner-text"
  >
    <h1
      className="banner-bold"
    >
      Dashboard
    </h1>
  </div>
  <div
    className="content"
  >
    <div
      className="playlist-container"
    >
      <div
        className="null playlist-card button-hover playlist-card-contents"
        key="0"
        onClick={[Function]}
        tabIndex="0"
      >
        <h5
          className="playlist-name-title"
        >
          playlist 1
        </h5>
      </div>
      <div
        className="null playlist-card button-hover playlist-card-contents"
        key="1"
        onClick={[Function]}
        tabIndex="0"
      >
        <h5
          className="playlist-name-title"
        >
          playlist 2
        </h5>
      </div>
    </div>
  </div>
</div>
`;
